src/PrimitiveObsession/Config/PistonCount.cs

```diff
@@ -1,10 +1,22 @@
 ﻿namespace PrimitiveObsession.Config
 {
-    public class PistonCount : ConfigurationValue<short>
-    {
-        public PistonCount(string value)
+    public class PistonCount
+    {
+        short Value { get; }
+
+        public PistonCount(short value)
         {
-            Value = short.Parse(value);
+            Value = value;
+        }
+
+        public static implicit operator short(PistonCount pistonCount)
+        {
+            return pistonCount.Value;
+        }
+
+        public static explicit operator PistonCount(short value)
+        {
+            return new PistonCount(value);

```

Comment:
Qui ho usato `implicit` e `explicit` invece di ereditare dalla classe custom `ConfigurationValue`.
La definizione delle classi viene un capellino pi\ lunga, ma l'effetto è che il casting al tipo corrispondente primitivo avviene in automatico.  La conversione avviene nei due versi, da primitivo a oggetto e viceversa.

Io lo trovo molto elegante.  Ma è solo una questione di stili e gusti.

Lo considero un commit opzionale e discutibile.